x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/index.tsx

```diff
@@ -681,39 +684,43 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
-                            rule,
-                            hasMlPermissions,
-                            hasActionsPrivileges,
-                            canUserCRUD
-                          )}
-                        />
-                      </EuiFlexItem>
-                      <EuiFlexItem grow={false}>
-                        <RuleActionsOverflow
-                          rule={rule}
-                          userHasPermissions={isExistingRule && hasUserCRUDPermission(canUserCRUD)}
-                          canDuplicateRuleWithActions={canEditRuleWithActions(
-                            rule,
-                            hasActionsPrivileges
-                          )}
-                          showBulkDuplicateExceptionsConfirmation={showBulkDuplicateConfirmation}
-                          showManualRuleRunConfirmation={showManualRuleRunConfirmation}
-                          confirmDeletion={confirmDeletion}
-                        />
-                      </EuiFlexItem>
-                    </EuiFlexGroup>
-                  </EuiFlexItem>
+                  {isSearchAiLake === false && (
+                    <EuiFlexItem grow={false}>
+                      <EuiFlexGroup alignItems="center" gutterSize="s" responsive={false}>
+                        <EuiFlexItem grow={false}>
+                          <EditRuleSettingButtonLink
+                            ruleId={ruleId}
+                            disabled={
+                              !isExistingRule ||
+                              !hasUserCRUDPermission(canUserCRUD) ||
+                              (isMlRule(rule?.type) && !hasMlPermissions)
+                            }
+                            disabledReason={explainLackOfPermission(
+                              rule,
+                              hasMlPermissions,
+                              hasActionsPrivileges,
+                              canUserCRUD
+                            )}
+                          />
+                        </EuiFlexItem>
+                        <EuiFlexItem grow={false}>
+                          <RuleActionsOverflow
+                            rule={rule}
+                            userHasPermissions={
+                              isExistingRule && hasUserCRUDPermission(canUserCRUD)
+                            }
+                            canDuplicateRuleWithActions={canEditRuleWithActions(
+                              rule,
+                              hasActionsPrivileges
+                            )}
+                            showBulkDuplicateExceptionsConfirmation={showBulkDuplicateConfirmation}
+                            showManualRuleRunConfirmation={showManualRuleRunConfirmation}
+                            confirmDeletion={confirmDeletion}
+                          />
+                        </EuiFlexItem>
+                      </EuiFlexGroup>
+                    </EuiFlexItem>
+                  )}
```

Comment:
Should rule actions and settings be extracted to a separate component for better readability?